network/src/filters/filter.rs

```diff
@@ -32,15 +38,22 @@ impl Default for Filter {
 }
 
 impl Filter {
-    pub fn new(input_vector: Vec<IpAddr>) -> Self {
+    pub fn new(input_vector: Vec<FilterEntry>) -> Self {
         Self {
             enabled: !input_vector.is_empty(),
-            list: input_vector.into_iter().collect(),
+            list: input_vector.into_iter().map(|x| (x.addr, x.tag)).collect(),
         }
     }
 
-    pub fn add(&mut self, addr: IpAddr) {
-        self.list.insert(addr);
+    pub fn add(&mut self, addr: IpAddr, tag: Option<String>) {
+        match tag {
```

Comment:
How about
```rust
self.list.insert(addr, tag.unwrap_or_default());
```

Reply:
I intended to not to change the tag value if the input is `None`.